src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/SlimefunMachine.java

```diff
@@ -2,18 +2,33 @@
 
 import java.util.*;
 
+import me.mrCookieSlime.CSCoreLibPlugin.general.Inventory.InvUtils;
 import me.mrCookieSlime.Slimefun.Lists.RecipeType;
 import me.mrCookieSlime.Slimefun.Objects.Category;
 import me.mrCookieSlime.Slimefun.Objects.MultiBlock;
+import me.mrCookieSlime.Slimefun.api.BlockStorage;
 
 import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.block.Chest;
+import org.bukkit.inventory.Inventory;
 import org.bukkit.inventory.ItemStack;
 
 public class SlimefunMachine extends SlimefunItem {
 	
 	private List<ItemStack[]> recipes;
 	private List<ItemStack> shownRecipes;
 	private Material trigger;
+	//Adjacent blockfaces for iterative output chest checks
+	private static final BlockFace[] faces = {
+		    BlockFace.DOWN,
+		    BlockFace.UP,
```

Comment:
I don't think we need UP and DOWN here.

Reply:
Well, UP is useful for stuff like the pressure chamber. Maybe not DOWN though, I'll have that removed.

Reply:
Yeah right, okay, keep UP but remove DOWN then.
DOWN does not make much sense since you cannot open the chest below a block anyway.